src/librustc/traits/trans/mod.rs

```diff
@@ -0,0 +1,73 @@
+use dep_graph::{DepGraph, DepNode, DepTrackingMap, DepTrackingMapConfig};
+use hir::def_id::DefId;
+use std::cell::RefCell;
+use std::marker::PhantomData;
+use traits::Vtable;
+use ty::{self, Ty};
+
+/// Specializes caches used in trans -- in particular, they assume all
+/// types are fully monomorphized and that free regions can be erased.
```

Comment:
I more or less want to kill all "trans" terminology and just have backends that use `Reveal::All` and do the lifetime erasing themselves, in the long term.